platform/darwin/src/MGLSymbolStyleLayer.h

```diff
@@ -6,67 +6,199 @@
 
 NS_ASSUME_NONNULL_BEGIN
 
+/**
+ Label placement relative to its geometry.
+ */
 typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerSymbolPlacement) {
+    /**
+     The label is placed at the point where the geometry is located.
+     */
     MGLSymbolStyleLayerSymbolPlacementPoint,
+    /**
+     The label is placed along the line of the geometry. Can only be used on `LineString` and `Polygon` geometries.
+     */
     MGLSymbolStyleLayerSymbolPlacementLine,
 };
 
+/**
+ In combination with `symbolPlacement`, determines the rotation behavior of icons.
+ */
 typedef NS_ENUM(NSUInteger, MGLSymbolStyleLayerIconRotationAlignment) {
+    /**
+     When `symbolPlacement` is set to `point`, aligns icons east-west. When `symbolPlacement` is set to `line`, aligns icon x-axes with the line.
```

Comment:
Done pending improvements in https://github.com/mapbox/mapbox-gl-native/pull/6508#discussion_r81259119.